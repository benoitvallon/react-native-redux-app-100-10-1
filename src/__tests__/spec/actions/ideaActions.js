'use strict';

import * as actions from '../../../actions/ideaActions';

describe('idea actions', function() {
  it('should be find an actions file', function() {
    expect(actions).toBeDefined();
  });

  it('should create an action to add an idea', function() {
    const idea = 'My first idea';
    expect(actions.add(idea)).toMatchSnapshot();
  });

  it('should create an action to save an idea', function() {
    const index = 1;
    const idea = 'My second idea';
    expect(actions.save(idea, index)).toMatchSnapshot();
  });

  it('should create an action to remove an idea', function() {
    const index = 1;
    expect(actions.remove(index)).toMatchSnapshot();
  });

  it('should create an action to reset the idea list', function() {
    expect(actions.reset()).toMatchSnapshot();
  });
});
