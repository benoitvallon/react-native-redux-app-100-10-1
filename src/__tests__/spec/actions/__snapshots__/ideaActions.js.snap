exports[`idea actions should create an action to add an idea 1`] = `
Object {
  "data": "My first idea",
  "type": "ADD",
}
`;

exports[`idea actions should create an action to remove an idea 1`] = `
Object {
  "data": 1,
  "type": "REMOVE",
}
`;

exports[`idea actions should create an action to reset the idea list 1`] = `
Object {
  "type": "RESET",
}
`;

exports[`idea actions should create an action to save an idea 1`] = `
Object {
  "data": Object {
    "idea": "My second idea",
    "index": 1,
  },
  "type": "SAVE",
}
`;
